### ISSUE TITLE
Can't mock properly a proxied plugin

### ISSUE THREAD

**Issue by WheelyWonka** [ionic-team/capacitor-plugins#615]
### Plugin(s)
@capacitor/app 1.0.3

### Capacitor Version

```
💊   Capacitor Doctor  💊

Latest Dependencies:

  @capacitor/cli: 3.2.3
  @capacitor/core: 3.2.3
  @capacitor/android: 3.2.3
  @capacitor/ios: 3.2.3

Installed Dependencies:

  @capacitor/android: not installed
  @capacitor/ios: not installed
  @capacitor/cli: 3.2.3
  @capacitor/core: 3.2.3
```

### Platform(s)
Karma / Jasmine / Angular

### Current Behavior
I can't manage to mock (spyOn) properly methods from App plugins, and globally every Capacitor's Plugins that are Proxied. 

I explain the detailed case in the [README.md of the reproduction repo](https://github.com/WheelyWonka/test-plugin-proxy-capacitor/tree/main).

As i put it in [the reproduction repo](https://github.com/WheelyWonka/test-plugin-proxy-capacitor/tree/main), I tried the [solution proposed here](https://github.com/ionic-team/cap-plugin-mock-jasmine) but with no success. It seems that I need to mock a lot of things to test one call to a method exposed by a Capacitor's Plugin. I find this way of doing test kinda weird, what happens if Capacitor is updating something I mocked in the first place, written tests will become wrong ? If someone have the time to clarify this philosophy to me, and how should I mock the `app.getInfo` method, it would be awesome.

### Expected Behavior
Be able to simply spyOn a plugin's method without deeply mocking related things to the method in order to avoid potentially drifts caused by future updates of the plugin.

### Code Reproduction
Here is the repo to reproduce the case here: https://github.com/WheelyWonka/test-plugin-proxy-capacitor/tree/main

Thank you for your time and your amazing work !

**Comment by Bator92** [comment]
Hello!

Did you make any progress with this issue? I'm facing a similar issue with the `Text Zoom` plugin and jest. ([https://capacitorjs.com/docs/apis/text-zoom](url)) I cannot find a way to spy on its methods.

I have this test:

```
 it('should call text zoom with value 1', () => {
    jest.spyOn(Capacitor, 'isNativePlatform').mockReturnValueOnce(true);
    const setZoomSpy = jest.spyOn(TextZoom, 'set');

    expect(setZoomSpy).toHaveBeenCalledWith({ value: 1 });
  });
```
I got the following error:
```
Error: expect(received).toHaveBeenCalledWith(...expected)

Matcher error: received value must be a mock or spy function

Received has type:  function
Received has value: [Function set]
```

I could mock the `Capacitor` plugin, but mocking the `TextZoom` plugin is not working. How do I know, that it is a proxied plugin?

**Comment by jfcere** [comment]
> I tried the [solution proposed here](https://github.com/ionic-team/cap-plugin-mock-jasmine) but with no success. It seems that I need to mock a lot of things to test one call to a method exposed by a Capacitor's Plugin. I find this way of doing test kinda weird, what happens if Capacitor is updating something I mocked in the first place, written tests will become wrong ?

I totally agree with that, creating our own mocks means we have to handle the changes when Capacitor updates the plugins.

One possible solution would be for Capacitor to provide those mocks so that they could be in sync with all the changes and development done on the plugins.

**Comment by eric-horodyski** [comment]
This thread has been inactive for some time, and we’re cleaning up older items to keep the backlog focused and manageable.

Details on the effort:
[https://ionic.io/blog/keeping-capacitors-backlog-healthy](https://ionic.io/blog/keeping-capacitors-backlog-healthy)

If this is still relevant today, feel free to open a new issue with current information and we’ll review it.
